shared/p2p/hobbits/hobbits.go

```diff
@@ -0,0 +1,64 @@
+package hobbits
+
+import (
+	"net"
+	"reflect"
+	"sync"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+	"github.com/prysmaticlabs/prysm/beacon-chain/db"
+	"github.com/prysmaticlabs/prysm/shared/p2p"
+	"github.com/renaynay/go-hobbits/encoding"
+	"github.com/renaynay/go-hobbits/tcp"
+	log "github.com/sirupsen/logrus"
+	ttl "github.com/ReneKroon/ttlcache"
+)
+
+type HobbitsNode struct {
+	sync.Mutex
+	NodeId       string
+	Host         string
+	Port         int
+	StaticPeers  []string
+	PeerConns    map[peer.ID]net.Conn
+	feeds        map[reflect.Type]p2p.Feed
+	Server       *tcp.Server
+	DB           *db.BeaconDB
+	MessageStore *ttl.Cache
+}
+
+type HobbitsMessage encoding.Message
+
+const CurrentHobbits = uint32(3)
+
+//var topicMapping map[reflect.Type]string // TODO: initialize with a const? How do I use this??
+
+type RPCMethod uint16
+
+const (
```

Comment:
Please add a link to hobbit spec that defines this structure. Wondering why it goes from iota order to iota + 8.

Reply:
can you shift all the constants to the top of the file, and then following it declare all the structs, types, etc